### ISSUE TITLE
When two independent Felixes run on the same bare metal

### ISSUE THREAD

**Issue by vzuevsky** [projectcalico/felix#3075]
I could do with some help to understand the following interaction.

We run `calico-felix` agent as root user with etcd backend on bare metal machines. It only process hostEndPoints and globalNetworkPolices objects to serve as a firewall in `Insert` mode.

I am trying to install k3s with calico cni on one of those machines. It runs `calico-node -felix`. I configured it to run in `Append` mode. The cluster was up and running, but then I noticed that the firewall `calico-felix` binary crashes. In fact it cannot run in parallel and crashes every time I start it.

I examined the log and main reason for crash is

> Failed to execute ip(6)tables-restore command error=exit status 2 errorOutput="iptables-restore v1.6.1: Set cali40s:yYdgw9xrU7F0ZsLcIdixX40 doesn't exist.\n\nError occurred at line: 64

And there are many such non-existing sets reported. I struggle to see how is that possible because the agent is to program those ipsets itself to ensure consistency. 

I also found in the logs that `calico-felix` agent tries to mess with `k3s`'s stuff, e.g.

> felix/route_table.go 861: Syncing routes: found unexpected route; ignoring due to grace period. dest=172.16.186.141/32 ifaceName="cali8ae45e324d6"

not sure what is "grace period" but what will happen beyond? And more troublesome:

> felix/table.go 685: Found unexpected chain, marking for cleanup chainName="cali-pro-kns.kube-system" ipVersion=0x4 table="filter"

My idea to separate the two by the mode - `Insert` vs `Append` - perhaps too simplistic and impractical. Especially if Felix relies on rule names in the comment exclusively without segregation by mode. What do you think?

**Comment by vzuevsky** [comment]
```
# /usr/local/bin/calico-felix --version
Version:            v3.17.1
Full git commit ID: 5e895d698874feffb5211a997b4ded8639dca53e
Build date:         2020-12-10T23:37:51+0000

# iptables --version
iptables v1.6.1

**Comment by caseydavenport** [comment]
Felix isn't designed to run two instances on the same host machine, they're likely to fight with each other as you've discovered over other resources like ipsets, routes, etc. 

> Alternatively, is it possible for a single instance of calico take policies from etcd datastore and concurrently use k8s datastore for anything else such as ipam?

Not today, you'd need to use a single data store with both your kubernetes and host policies in it.

**Comment by vzuevsky** [comment]
Thanks for your advice, Casey. We have a central etcd cluster containing globalNetworkPolicies, hostEndPoints, and Node objects. Those are solely used for firewall'ing purpose. The key prefixes used in etcd store by calico-felix binary are
```
/calico/resources/v3/projectcalico.org/hostendpoints
/calico/resources/v3/projectcalico.org/clusterinformations
/calico/resources/v3/projectcalico.org/nodes
/calico/resources/v3/projectcalico.org/globalnetworkpolicies
```
There was a wording in the documentation recommending to NOT use the same store for kubernetes registries. Technically, I can point `calico-node -felix` process of the cluster to our global etcd store. Vanilla, each k3s cluster uses its local store; if I point many clusters to the global store:

1/ can there be conflicts between clusters?
2/ will cluster configs mess with those prefixes above?
3/ is it a recommended practice to keep global network polices and local cluster configs in one central store?

**Comment by caseydavenport** [comment]
> 1/ can there be conflicts between k3s clusters?

Yeah - you definitely want a separate datastore for each cluster. Most folks today run using the "kubernetes API datastore" which doesn't require a separate etcd cluster, and instead uses CRDs in the Kubernetes API server for config storage.

"global network policies" aren't global in the sense that they apply to multiple clusters, rather they are cluster-wide.
